src/lua/init.lua

```diff
@@ -211,12 +211,18 @@ local mt = {
     __serialize = filter_out_private_fields,
 }
 
-return setmetatable({
+local extension = {
     uptime = uptime,
     pid = pid,
     _internal = {
         strip_cwd_from_path = strip_cwd_from_path,
         module_name_from_filename = module_name_from_filename,
         run_preload = run_preload,
     },
-}, mt)
+}
+-- tarantool module is already registered by src/lua/init.c
+local tarantool = require('tarantool')
```

Comment:
AFAIU, by our convention, the internal module should be named `tarantool.lib`.

Reply:
THere are two ways to express a module that is partially implemented in C and partially implemented in Lua.

1. `foo.lib` (C) + `foo` (Lua). In this case `foo` exports needed `foo.lib` functions directly or using a wrapper function. Example: `socket`.
2. `foo` (C) + `foo` (Lua). In this case `foo` MUST return the same module table, but it can add it more functions. Example: `fio`.

Both ways looks OK for me.

We discussed it f2f with Nikolay. He prefers the second way, because the functions implemented in C are exposed directly, without any wrapper. This reason also seems OK to me.

Reply:
Moved `require('tarantool')` to the top of the file as disscussed f2f. Also used `tarantool_lua` instead of `extension`.